src/note_position.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt;
+use std::iter::Peekable;
+
+/// An iterator adaptor that wraps each element in an [`Position`](../enum.Position.html).
+///
+/// Iterator element type is `Position<I::Item>`.
+/// This iterator is *fused*.
```

Comment:
hm ok, after some thinking about peekable, it's actually pretty bad that it's not fused.

Calling `.peek()` and observing `None` is the only signal the user gets that the iterator is exhausted -- calling `.next()` after that leads to iterator-specific behaviour.